ui/src/criteria/attribute.tsx

```diff
@@ -81,23 +141,149 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+  const enumHintValues = attributeFilterHint?.enumHint?.enumHintValues;
+  const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
 
-  const selectionIndex = (hint: tanagra.EnumHintValue) =>
-    props.data.selected.findIndex(
-      (row) => row.id === hint.attributeValue?.int64Val
-    );
+  console.log(underlay);
+  console.log(attributeFilterHint);
 
-  if (enumHintValues?.length === 0) {
+  if (!isValid(integerBoundsHint) && enumHintValues?.length === 0) {
     return (
       <Typography>
         No information for attribute {props.data.attribute}.
       </Typography>
     );
   }
+
+  if (
+    integerBoundsHint?.max !== null &&
+    integerBoundsHint?.max !== undefined &&
+    integerBoundsHint?.min !== null &&
+    integerBoundsHint?.min !== undefined
+  ) {
+    const minBound = integerBoundsHint?.min;
```

Comment:
This is complicated enough that it's definitely worth splitting into a subcomponent. If it's not going to be used anywhere else, defining it privately in this file is fine - alternatively it could go into a components directory. 


```
function IntegerRangeSlider({...}: IntegerRangeSliderProps) {
  ...
}

...
if (hasValidIntegerBounds) {
  return <IntegerRangeSlider ...>;
}
```

Reply:
Thank you CH! The compiler always complain that `integerBoundsHint.min` and `integerBoundsHint.max` might be undefined. This explicit check was here to tell the compiler that min and max are valid. I improve the code by using your method and also put comments here to explain this.